profile.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package eat
+
+import (
+	"strconv"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"encoding/asn1"
+	cbor "github.com/fxamacker/cbor/v2"
+)
+
+// Profile is either a well formed RFC3986 compliant URI(uri  *url.URL).
+// or an ObjectIdentifier  ([]int)
+type Profile struct {
+	val interface{}
+}
+
+// DecodeProfileCBOR decodes from a received CBOR data the profile
+// as either a URL or a Object Identifier
+func (s Profile) DecodeProfileCBOR(val interface{}) error
+{
+	switch t := val.(type) {
+	case string:
+		if len(val) == 0 {
+			return fmt.Errorf("No valid URL for profile")
+		}
+		s.val, err := url.Parse(s.val)
+		if err != nil {
+			return fmt.Errorf("Failed to parse profile URL: %w", err)
+		}
+	case []byte:
+		if len(val) == 0 {
+			return fmt.Errorf("No valid OID for profile")
+		}
+		_, err := asn1.Unmarshal(val, &s.val)
+		if (err != nil) {
+			return fmt.Errorf("MalFormed profile OID detetced")
+		}
+	}
+} 
+// MarshalCBOR will encode the Profile value either as a CBOR text string(URL),
+// or as byte array 
+func (s Profile) MarshalCBOR() ([]byte, error) {
+	switch t := s.val.(type) {
+	case *url.URL:
+		var uri *url.URL
+		uri = s.val.(*url.URL)
+		return em.Marshal(uri.String())
+
+	case asn1.ObjectIdentifier:
+		var asn1_oid []byte
+
+		asn1_oid, err := asn1.Marshal(s.val)
+		if err != nil{
+			return nil, fmt.Errorf("ASN1 Encoding failed for OID", err)
+		}
+		return em.Marshal(asn1_oid)
+	}
```

Comment:
maybe catch the default condition?

Reply:
ok, added a default condition in the code!